localgov_directories.install

```diff
@@ -84,3 +87,154 @@ function localgov_directories_update_8002() {
+    if (!NodeType::load($bundle)) {
+      continue;
+    }
+    if (!FieldConfig::loadByName('node', $bundle, 'localgov_directory_title_sort')) {
+      FieldConfig::create([
+        'field_name' => 'localgov_directory_title_sort',
+        'entity_type' => 'node',
+        'bundle' => $bundle,
+        'label' => 'Title used for sorting',
+        'description' => "<strong>Can be left blank</strong>. If this field is completed it will be used instead of the <em>Title</em> for alphabetically sorted lists. For example to move 'The' or 'A' from the beginning of a name.",
+        'required' => FALSE,
+      ])->save();
+
+      $result[] = t('Added localgov_directory_title_sort to %bundle.', ['%bundle' => $bundle]);
+    }
+    else {
+      $result[] = t('Not added localgov_directory_title_sort to %bundle as it already exists.', ['%bundle' => $bundle]);
+    }
+
+    // Add to form display.
+    $form_display = EntityFormDisplay::load('node.' . $bundle . '.default');
+    $form_display->setComponent('localgov_directory_title_sort', [
+      'weight' => 1,
+      'settings' => [
+        'size' => 60,
+        'placeholder' => '',
+      ],
+      'third_party_settings' => [],
+      'type' => 'string_textfield',
+      'region' => 'content',
+    ]);
+    // Place under title field.
+    $form_group_description = $form_display->getThirdPartySetting('field_group', 'group_description');
+    if (!empty($form_group_description) &&
+        (($title_position = array_search('title', $form_group_description['children'])) !== FALSE)
+    ) {
+      array_splice($form_group_description['children'], $title_position + 1, 0, 'localgov_directory_title_sort');
+      $form_display->setThirdPartySetting('field_group', 'group_description', $form_group_description);
+      foreach ($form_group_description['children'] as $weight => $component_name) {
+        $component = $form_display->getComponent($component_name);
+        $component['weight'] = $weight;
+        $form_display->setComponent($component_name, $component);
+      }
+    }
+    else {
+      $result[] = t('Unable to set position of new sort field under title field on form display for %bundle', ['%bundle' => $bundle]);
+    }
+    $form_display->save();
+    $result[] = t('Updated form display for %bundle.', ['%bundle' => $bundle]);
+  }
+
+  return implode(' ', $result);
+}
+
+/**
+ * Add new sort field to search index.
+ */
+function localgov_directories_update_8005() {
+  // Add to the search index.
+  $index = Index::load('localgov_directories_index_default');
+  if (!$index->getField('localgov_directories_index_default')) {
+    $sort_title_field = new Field($index, 'localgov_directory_title_sort');
+    $sort_title_field->setLabel('Title (sort)');
+    $sort_title_field->setDatasourceId('entity:node');
+    $sort_title_field->setType('string');
+    $sort_title_field->setPropertyPath('localgov_directory_title_sort');
```

Comment:
We may need to set the dependency here.  Something like the following.  If you please have a look.
```
$sort_title_field->setDependencies([
  'config' => [
    "field.storage.node.localgov_directory_title_sort",
  ],
]);
```

Reply:
Can you run this, and do a `drush cex` and check but I'm pretty certain it does this itself.

Reply:
You are right.  The dependency is automatically added.  Ignore what I have said.